Voice Trigger Multitool Linkable

LukaSlade opened the pull request:
## About the PR
Allows the voice trigger to be linked to objects via multitool and can trigger signals upon recorded key phrases.

## Why / Balance
Changed for the better fun! Don't think it would be too unbalanced but also could definitely allow for some really interesting creations for both crew and antagonists.

## Technical details
Makes the voice trigger a connection source, and adds the voice recording component to the device itself without needing to be within a modular grenade.

## Requirements
- [X] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html)
- [x] I have added media to this PR or it does not require an ingame showcase.

Door Video Demo:
https://github.com/user-attachments/assets/7f7b163f-7724-4a6a-87d4-a04c40969264

Other Video Demo:
https://github.com/user-attachments/assets/88c40c22-2452-4733-a82f-9bf5744cbc4a

**Changelog**
:cl: PickelSilly
- tweak: Voice trigger tweaked to be its own standalone device and linkable to other objects using a multitool while still being a payload trigger for modular grenades.

iaada commented:
This same functionality can already be done by putting a remote signaler in a modular grenade, so cutting out the middleman makes sense to me.

slarticodefast commented:
This does make the signal trigger obsolete as an item ad the same effect can already be achieved with that, so I would rather keep them separate.

slarticodefast commented:
Nevermind, I mixed up the source and the sink, this way around it makes sense
my bad

LukaSlade commented:
Fixed some stuff, changed sound to a click sound, added a delay, gave it a proper component to make sure the sink it’s triggering is actually in range of it. I also made it use the source Trigger instead of Pressed to make it more accurate.

iaada commented:
This is only an idea but I'd like to see the Advanced Remote Signaler reworked to accept these triggers instead of being just a simply better remote signaler. They're in the same research, it would give the item a unique appeal, and give a little bit more support to the family of modular triggers/payloads.

LukaSlade commented:
Yeah im gonna go ahead and make a new PR for this, gonna make it a new separate item available in the construct menu or smth, seems more fitting.
